app/javascript/components/Churn/ChurnChart.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from "react";
+import { Area, Line, XAxis, YAxis } from "recharts";
+
+import { formatPriceCentsWithCurrencySymbol } from "$app/utils/currency";
+
+import useChartTooltip from "$app/components/Analytics/useChartTooltip";
+import { Chart, lineProps, xAxisProps, yAxisProps } from "$app/components/Chart";
+
+type DotProps = {
+  key: string;
+  cx: number;
+  cy: number;
+  width: number;
+};
+
+export type DataPoint = {
+  date: string;
+  label: string;
+  title: string;
+  churnRate: number;
+  churnedCustomers: number;
+  revenueLostCents: number;
+};
+
+const ChartTooltip = ({ data }: { data: DataPoint }) => (
+  <>
+    <div>
+      <strong>{data.churnRate.toFixed(2)}%</strong> churn
+    </div>
+    <div>
+      <strong>{data.churnedCustomers}</strong> {data.churnedCustomers === 1 ? "cancellation" : "cancellations"}
+    </div>
+    <div>
+      <strong>
+        {formatPriceCentsWithCurrencySymbol("usd", data.revenueLostCents, {
+          symbolFormat: "short",
+          noCentsIfWhole: true,
+        })}
+      </strong>{" "}
+      revenue lost
+    </div>
+    <time className="block font-bold">{data.title}</time>
+  </>
+);
+
+export const ChurnChart = ({ data }: { data: DataPoint[] }) => {
+  const { tooltip, containerRef, dotRef, events } = useChartTooltip();
+  const tooltipData = tooltip ? data[tooltip.index] : null;
+
+  const hasNonZero = data.some((point) => point.churnRate > 0);
+  const yDomain: [number, number | "dataMax"] = hasNonZero ? [0, "dataMax"] : [0, 1];
+
+  return (
+    <Chart
+      containerRef={containerRef}
+      tooltip={tooltipData ? <ChartTooltip data={tooltipData} /> : null}
+      tooltipPosition={tooltip?.position ?? null}
+      data={data}
+      maxBarSize={40}
+      margin={{ top: 32, right: 0, bottom: 16, left: 16 }}
+      {...events}
+    >
+      <XAxis {...xAxisProps} dataKey="label" />
+      <YAxis {...yAxisProps} domain={yDomain} tickFormatter={(value: number) => `${Number(value.toFixed(2))}%`} />
+      <Area dataKey="churnRate" fill="rgb(var(--info))" fillOpacity={0.15} stroke="none" isAnimationActive={false} />
```

Comment:
The fill color for the area in [Figma](https://www.figma.com/design/3ZTqynf6cCoaSuFDnF2KDo/Analytics?node-id=249-14259&t=FwSgz13OPcyB6QOy-0) is `border/info/subtle`. Using `--info` with opacity `0.15` seems close enough, @laugardie let me know if you would like to use something else.

**Figma**
<img width="1218" height="1074" alt="CleanShot 2025-12-28 at 08 27 59@2x" src="https://github.com/user-attachments/assets/c256ce1d-c715-408f-8167-abc0fc25146d" />

**App**
<img width="2628" height="1178" alt="CleanShot 2025-12-28 at 08 34 13@2x" src="https://github.com/user-attachments/assets/d1f29678-4b2e-495f-a7f8-1eb1b515ac44" />